Vizzuality/heco-invest
# feat: Impact calculation service

This MR brings new `Imports::CalculateForProject ` service, which is responsible for calculation of all impacts for all impact types and its dimensions. I have basically rewritten logic used at https://docs.google.com/spreadsheets/d/1KFYcduykY_mst41UKgafSAaHFvxp68wMRR0zGfdQkDs/edit#gid=1744503386 and all tests of this service are based on results computed by this sheet. Please @ikerey , I will add you as reviewer to this PR, so if you could just briefly check this service if you see some issues (hopefully, it is readable enough even for non-ruby person). New columns of Project used to store computed impacts are:
 - municipality_biodiversity_impact
 - municipality_climate_impact
 - municipality_water_impact
 - municipality_community_impact
 - municipality_total_impact
 - hydrobasin_biodiversity_impact
 - hydrobasin_climate_impact
 - hydrobasin_water_impact
 - hydrobasin_community_impact
 - hydrobasin_total_impact
 - priority_landscape_biodiversity_impact
 - priority_landscape_climate_impact
 - priority_landscape_water_impact
 - priority_landscape_community_impact
 - priority_landscape_total_impact

I save computed results as decimals, but from design, it looks like it will be in the end multiplied by `100 (to get %)` and rounded -- can be always done later on or by FE itself. All these variables are already part of Project API serializer and based on these values, sorting of Projects can be done.

I have slightly tweaked `backend/app/models/impact_area.rb` enums -- addition of `hydrometerological-risk-reduction` and `water-capacity` and `water-efficiency` were merged to one option called `water-capacity-or-efficiency`.

This MR also replaces `latitude` and `longitude` columns of Project model with `center` column which is of `point` type (lets use postgis when we already have it :)).

## Testing instructions

`ImpactCalculationJob ` should be automatically enqueued when you modify `geometry` or `impact_areas` of Project. Please @agnessa could you check if queue of `ImpactCalculationJob ` is correct, I am not really sure what should be there or if jobs already work at Stage. After job finishes, new impact attributes at Project record should be filled with correct values.

My proposal how to tests this would be following:
 - open spreadsheet and setup demands and impacted areas of project some random way
 - find some suitable `project` at db and setup `impact_areas` to same values as in spreadsheet
 - find location which this project intersect with and setup `demand` attributes based on spreadsheet
 - run `ImpactCalculationJob ` job and after it finishes, make sure that impact attributes of `project` correspond to values at spreadsheet

## Tracking

https://vizzuality.atlassian.net/browse/LET-382
https://vizzuality.atlassian.net/browse/LET-429
